src/nvim/path.c

```diff
@@ -407,8 +407,9 @@ void add_pathsep(char *p)
 /// @param fname is the filename to save
 /// @param force is a flag to expand `fname` even if it looks absolute
 ///
-/// @return [allocated] Copy of absolute path to `fname` or NULL when
-///                     `fname` is NULL.
+/// @return [allocated]
+///   - Copy of absolute path to `fname` (truncated if longer than MAXPATHL)
+///   - NULL when `fname` is NULL or `vim_FullName()` fails
```

Comment:
No. `retval=FAIL` if `path_get_absolute_path()` returns that. With your changes the behavior has changed.